python/beeai_framework/adapters/litellm/chat.py

```diff
@@ -55,14 +58,39 @@ class LiteLLMChatModel(ChatModel, ABC):
     def model_id(self) -> str:
         return self._model_id
 
+    @property
+    def provider_id(self) -> ProviderName:
+        """The provider of the underlying LLM. e.g. 'openai'"""
+        raise NotImplementedError()
+
     def __init__(self, model_id: str, *, provider_id: str, settings: dict | None = None) -> None:
         super().__init__()
         self._model_id = model_id
         self._litellm_provider_id = provider_id
         self.supported_params = get_supported_openai_params(model=self.model_id, custom_llm_provider=provider_id) or []
         # drop any unsupported parameters that were passed in
         litellm.drop_params = True
-        self._settings = settings or {}
+        self._settings = settings.copy() if settings is not None else {}
+
+        # Extra headers logic
+        extra_headers = None
+        extra_headers_json = self._settings.get("extra_headers", os.getenv("LITELLM_EXTRA_HEADERS"))
+
+        if extra_headers_json:
+            try:
+                parsed_headers = json.loads(extra_headers_json)
+                if isinstance(parsed_headers, dict):
+                    extra_headers = parsed_headers
```

Comment:
The parsing does not seem correct. In `TS` the input is not JSON, but a string with commas (KEY=VALUE,KEY2=VALUE2).

Reply:
Agreed, in that it was intentionally written to be a json string. As you point out this differs from typescript.

Headers can get complex, and need escaping if we want to handle the most general case, so going with a well defined format (like json) would seem to be more appropriate?

Reply:
it will also mean the env var will get set to the same value as the extra_headers property used by the ChatModel.

Reply:
In team call 20250312 consensus was that the extra headers should use simple strings, not json - will update

Reply:
* Updated implementation to use key value pairs.
* Tested with rits
* Error checking/parsing is simple, just checking for '='. Any issues will be met with a log entry, and ignoring the value set.
* No change to how extra_headers property is used. This remains in JSON format.